### DISCUSSION TITLE
Router codegen: inherit local Prettier config

### DISCUSSION THREAD

**Discussion by a-type** [TanStack/router#2264] [Ideas]
Presently, when running my Vite app with the router plugin, the route tree seems to continually be regenerated on file changes, writing to a format which doesn't match my Prettier config (which uses tabs, not spaces). Since Prettier runs on committed files automatically via a git hook in my project, if I attempt to commit this change, the route tree file is then rewritten using my project's Prettier config, returning it to its original state, leading to no working changes in Git again, and the commit fails.

The end result is my git repo always has dirty changes on the route tree file which can never be committed, only endlessly reverted to keep the working changes clean.

My workaround is to add the route tree file to my `.prettierignore`, which is probably intended, but I think it would be preferable for the generator to utilize Prettier's config discovery to simply inherit the closest prettier config if available, rather than exposing some subset of options to the plugin config (or, in addition to that).

**Comment by schiller-manuel** [comment]
you are correct, the file should be ignored by any formatter.
This is why we have added a header that starts with 

```tsx
/* prettier-ignore-start */
```

Formatting the generated file using those prettier options is deprecated and will be removed in future major releases.

**Reply by a-type** [reply]
Ah, I see. It doesn't appear to work; both my editor's Prettier-on-save and running `prettier --write` on the file both modify the file.

From the Prettier docs, it's a little ambiguous, but it looks like the `ignore-start/end` blocks only work in Markdown, for some reason. It's only listed as a subheading of the Markdown section: https://prettier.io/docs/en/ignore#range-ignore

I added the file to a top-level `.prettierignore` in my project and things are now working well. Notably, I tried a `.prettierignore` in the React project's root (this is a monorepo) and that prevented my editor from prettifying it on save, but it did not prevent `lint-staged` from doing so (I assume because `lint-staged` is running Prettier in some more obtuse way that fails to recognize the ignore file in a subdirectory, but does at the root of the repo...).

**Reply by rstrand** [reply]
> From the Prettier docs, it's a little ambiguous, but it looks like the `ignore-start/end` blocks only work in Markdown, for some reason. It's only listed as a subheading of the Markdown section: https://prettier.io/docs/en/ignore#range-ignore

prettier-ignore-start/end definitely only work in markdown. There is an open issue in the prettier github asking for support for other languages. https://github.com/prettier/prettier/issues/5287

**Comment by IanVS** [comment]
When using file-based routing, and moving things around for instance, it also formats my code poorly.  Would love if it used my prettier config instead.
